src/main/java/seedu/apollo/command/module/AddModuleCommand.java

```diff
@@ -99,6 +89,28 @@ public void execute(TaskList taskList, Ui ui, Storage storage, ModuleList module
         }
     }
 
+    /**
+     * Handles the case where the user wants to add a module.
+     *
+     * @param moduleList The list of modules.
+     * @param allModules Module backend data.
+     * @param ui The user interface.
+     * @throws DuplicateModuleException If the module already exists.
+     */
+    private void handleSingleCommand(ModuleList moduleList, ModuleList allModules, Ui ui)
+            throws DuplicateModuleException {
+        if (isAdded(moduleList, module)) {
+            throw new DuplicateModuleException();
+        }
+
+        if (module != null) {
```

Comment:
Perhaps you can extract another method for line 106 to 110/111, void addAndSortModule for example. More SLAP!